### PR TITLE
Feature lock move

### PR DESCRIPTION
**Pull request by phgage**
1. Database.java - refactored and added methods to get data from users and accounts db and create corresponding customer and account objects. Refactored Customer and Account classes
2. Removed Hashmap customers
3. Refactored code in NewBankClientHandler, NewBank for login and locking account
4. Started to refactor AccountManagement and Transaction classes - moveFunds now working however other methods not yet refactored
5. Port 14002 is now a static final variable

Did what I could last night/this morning, there is still a fair bit of refactoring left however it seems to work. Have not tested extensively so am sure there will be some holes
Thanks let me know what you think

### CODE REVIEW

File: server/Customer.java [hunk 1]
```diff
@@ -65,12 +85,19 @@ public String getPassword() {
...
 
-	public void setAccountLocked(boolean value){
-		accountLocked = value;
-	}
-
 	public boolean isAccountLocked() {
 		return accountLocked;
 	}
 
+	public void setAccountLocked(boolean bool){
```
Comment (MattKomorcec) [inline server/Customer.java]: You shouldn't call a parameter or a variable `bool`. For example, in C#, `bool` is the data type, so it's a really bad practice to do this. Same goes with `string`, `int` etc.

File: server/Customer.java [hunk 2]
```diff
@@ -65,12 +85,19 @@ public String getPassword() {
...
+		accountLocked = bool;
+		int value = (bool) ? 1 : 0;
+		try {
+			database.setLockAccount(username, value);
+		}catch (Exception e){
+			System.out.println("EXCEPTION!! Database.java: " + e.getMessage());
```
Comment (MattKomorcec) [inline server/Customer.java]: This is not `Database.java` file :)

File: server/Database.java [hunk 3]
```diff
@@ -28,38 +28,79 @@
...
+            while (results.next()) {
+                // Gets all the values from the results of running the query
+                results.getInt("id");
+                String accountNumber = results.getString("account_number");
+                String accountType = results.getString("account_type");
+                int balance = results.getInt("balance");
+                String sortCode = results.getString("sortcode");
+                int userID = results.getInt("user_id");
+
+                // Creates an object of a customer class, using the retrieved values
```
Comment (MattKomorcec) [inline server/Database.java]: This comment should be removed

File: server/Database.java [hunk 4]
```diff
@@ -28,38 +28,79 @@
...
+            String query = "SELECT * FROM accounts WHERE user_id = ? ";
+            PreparedStatement statement = conn.prepareStatement(query);
+            statement.setInt(1, userNumber);
+
+            // Executes the query, and retrieves the results
+            ResultSet results = statement.executeQuery();
+
+            while (results.next()) {
+                // Gets all the values from the results of running the query
+                results.getInt("id");
```
Comment (MattKomorcec) [inline server/Database.java]: You didn't add `id` to the `Account` class, so no need to retrieve it here

File: server/Database.java [hunk 5]
```diff
@@ -28,38 +28,79 @@
...
+                String accountNumber = results.getString("account_number");
+                String accountType = results.getString("account_type");
+                int balance = results.getInt("balance");
+                String sortCode = results.getString("sortcode");
+                int userID = results.getInt("user_id");
+
+                // Creates an object of a customer class, using the retrieved values
+                Account account = new Account(accountNumber, accountType, balance, sortCode, userID);
+                accounts.add(account);
+                System.out.println(accountNumber);
```
Comment (MattKomorcec) [inline server/Database.java]: If you want to keep these, just put them below inside of the `if (debug)`

File: server/Transaction.java [hunk 6]
```diff
@@ -229,12 +247,12 @@ private boolean checkDouble(String s) {
     }
 
     // Returning true if there are sufficient funds, else false.
-    private boolean checkSufficientFunds(double amount, Account fromAccount) {
+    private boolean checkSufficientFunds(int amount, Account fromAccount) {
         return !((fromAccount.getBalance() - amount) < 0);
```
Comment (MattKomorcec) [inline server/Transaction.java]: That's a really interesting way of writing this, took me a bit to realize what was the point!
Doesn't `fromAccount.getBalance() - amount > 0` feel more natural?

**Review by MattKomorcec — COMMENTED** [review]
Some small comments

**Review by portiamcd1 — COMMENTED** [review]
Good Evening, 

I just had a look through the commits. I can see you have addressed the previous comments. 

The only very small thing I can see is that some catches have an empty line before the finally, but others don't throughout the code. Not a major thing at all - I just wanted to understand if that was intentional or not? :) 

Apart from that, I cannot see any further changes needed at all - thanks for the hard work! :)

**Comment by phgage** [comment]
Thanks both, most of the errors you mentioned are usually because I write the first thing that comes to mind, I need to review my changes more comprehensively!

Mat, can I ask a question I forgot to ask about the Database class - what do the statements with @ in the comments do? Like @param 

Portia - just checking do you mean whitespace formatting or an empty catch statement? I have tried to address the former...

Thanks all

S

**Comment by portiamcd1** [comment]
Hi Stephen, I can see you have addressed that now - sorry for any confusion :)

**Pull request merged**